src/Infrastructure/Repository/CampoKeysRepository.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Fau\DegreeProgram\Common\Infrastructure\Repository;
+
+use Fau\DegreeProgram\Common\Domain\CampoKeys;
+use Fau\DegreeProgram\Common\Domain\DegreeProgram;
+use Fau\DegreeProgram\Common\Domain\DegreeProgramId;
+use Fau\DegreeProgram\Common\Infrastructure\Content\Taxonomy\AreaOfStudyTaxonomy;
+use Fau\DegreeProgram\Common\Infrastructure\Content\Taxonomy\DegreeTaxonomy;
+use Fau\DegreeProgram\Common\Infrastructure\Content\Taxonomy\StudyLocationTaxonomy;
+use WP_Error;
+use WP_Term;
+
+final class CampoKeysRepository
+{
+    public const CAMPO_KEYS_TOTAXONOMY_MAP = [
+        DegreeTaxonomy::KEY => DegreeProgram::DEGREE,
+        StudyLocationTaxonomy::KEY => DegreeProgram::LOCATION,
+        AreaOfStudyTaxonomy::KEY => DegreeProgram::AREA_OF_STUDY,
+    ];
+
+    public const CAMPOKEY_TERM_META_KEY = 'uniquename';
+
+    public function degreeProgramCampoKeys(DegreeProgramId $degreeProgramId): CampoKeys
+    {
+        $campoKeys = CampoKeys::empty();
+
+        /** @var WP_Error|array<WP_Term> $terms */
+        $terms = wp_get_post_terms(
+            $degreeProgramId->asInt(),
+            array_keys(self::CAMPO_KEYS_TOTAXONOMY_MAP)
+        );
+
+        if ($terms instanceof WP_Error) {
+            return $campoKeys;
+        }
+
+        foreach ($terms as $term) {
+            $campoKey = (string) get_term_meta($term->term_id, self::CAMPOKEY_TERM_META_KEY, true);
+
+            if (empty($campoKey)) {
+                continue;
+            }
+
+            $campoKeyType = self::CAMPO_KEYS_TOTAXONOMY_MAP[$term->taxonomy] ?? null;
+
+            if (is_null($campoKeyType)) {
+                continue;
+            }
+
+            $campoKeys->set($campoKeyType, $campoKey);
+        }
+
+        return $campoKeys;
+    }
+
+    /**
+     * Return a map of taxonomy keys to terms based on a given HIS code.
+     *
+     * @return array<string, int>
+     */
+    public function taxonomyToTermsMapFromCampoKeys(CampoKeys $campoKeys): array
+    {
+        $result = [];
+
+        $campoKeys = $campoKeys->asArray();
+
+        foreach (self::CAMPO_KEYS_TOTAXONOMY_MAP as $taxonomy => $campoKeyType) {
+            $campoKey = $campoKeys[$campoKeyType] ?? '';
+
+            if ($campoKey === '') {
+                continue;
+            }
+
+            $term = $this->findTermByCampoKey($taxonomy, $campoKey);
+            $result[$taxonomy] = ! is_null($term) ? $term->term_id : 0;
```

Comment:
I would not add an empty value to the result.

Reply:
the problem is if we don't pass `0` term id it will continue finding results with wrong HIS codes.

Reply:
I meant something like
```php
if ($term instanceof \WP_Term) {
   $result[$taxonomy] = $term->term_id;
}
```

Does it work?

Reply:
the problem still remains the same, consider a scenario where editor passes an HIS code like `185|80`
but there is no degree with campo key `185` (a.k.a the `$term` here is null) and as a result the tax query item for degree isn't built [here](https://github.com/RRZE-Webteam/FAU-Studium-Common/blob/0816b3af8d637e943e0d8cd1ea1f80349f19a79e/src/Infrastructure/Repository/WpQueryArgsBuilder.php#L103) so the tax query doesn't include degree at all and will lead to wrong results, while it shouldn't show anything if a part of HIS code is wrong.

Reply:
Oh, okay. So, we create an invalid WP_Query to enforce empty results. We could throw and catch at a higher level the exception instead. On the other side, we don't validate other filters/terms IDs... Could you add a comment to explain this?

Reply:
sure that makes more sense, I followed the exception approach with a clarifying comment.